common/psbt_keypath.c

```diff
@@ -0,0 +1,33 @@
+#include "config.h"
+#include <bitcoin/tx.h>
+#include <common/psbt_keypath.h>
+#include <common/utils.h>
+#include <wally_bip32.h>
+#include <wally_psbt.h>
+
+void psbt_set_keypath(u32 index, const struct ext_key *ext, struct wally_map *map_in) {
+	u8 fingerprint[BIP32_KEY_FINGERPRINT_LEN];
+	if (bip32_key_get_fingerprint(
+		    (struct ext_key *) ext, fingerprint, sizeof(fingerprint)) != WALLY_OK)
+		abort();
+
+	u32 path[1];
+	path[0] = index;
+
+	if (wally_map_add_keypath_item(map_in,
+				       ext->pub_key, sizeof(ext->pub_key),
+				       fingerprint, sizeof(fingerprint),
+				       path, 1) != WALLY_OK)
+		abort();
+}
+
+void psbt_add_keypath_to_last_output(struct bitcoin_tx *tx,
+				     u32 index,
```

Comment:
rename to `key_index` or `child_index` to clarify?

Reply:
done